src/libraries/System.Text.Json/src/System.Text.Json.csproj

```diff
@@ -211,43 +211,55 @@
     <Compile Include="System\Text\Json\Writer\Utf8JsonWriter.WriteValues.String.cs" />
     <Compile Include="System\Text\Json\Writer\Utf8JsonWriter.WriteValues.UnsignedNumber.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsNetStandard)' == 'true' or '$(TargetsNetFx)' == 'true'">
-    <Compile Include="System\Collections\Generic\StackExtensions.netstandard.cs" />
-    <!-- Common or Common-branched source files -->
-    <Compile Include="$(CommonPath)System\Buffers\ArrayBufferWriter.cs"
-             Link="Common\System\Buffers\ArrayBufferWriter.cs" />
+  <ItemGroup Condition="$(TargetFramework.StartsWith('net4'))">
+    <Reference Include="Microsoft.Bcl.AsyncInterfaces" />
     <Reference Include="mscorlib" />
     <Reference Include="System" />
     <Reference Include="System.Core" />
     <Reference Include="System.ValueTuple" />
+    <Reference Include="System.Buffers" />
+    <Reference Include="System.Memory" />
     <Reference Include="System.Numerics.Vectors" />
-    <Reference Include="Microsoft.Bcl.AsyncInterfaces" />
+    <Reference Include="System.Threading.Tasks.Extensions" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
```

Comment:
Just verifying, are these packaging changes correct and expected, @steveharter, @layomia?

Reply:
These are infra changes only. Instead of bootstrapping and copying these referenced packages upfront we now just PackageReference them.

Reply:
Did you consider making this PR switch everything other than NetCoreAppCurrent to PackageReference?  That might avoid having to add new conditions in most places.

Reply:
I tried that with https://github.com/dotnet/runtime/pull/34893 but it became convoluted. Especially with $(NetFrameworkCurrent) which we run tests against. I'm fine with the extra costs for incremental progress.